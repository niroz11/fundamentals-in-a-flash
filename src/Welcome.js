import React, { Component } from 'react';

class Welcome extends Component {
  constructor(props) {
    super(props);
    this.state = {
      
    };
  }
  updateCategory = (event) => {
    event.preventDefault()
    this.props.updateCategory(event.target.innerHTML)
  }
  render() {
    return (
      <div className="welcome">
        <p>Take these short quizes to solidify your JS Fundamentals knowledge</p>
        <p>Click a category below to get started</p>
        {
          this.props.categories.map((category, index) => {
            return <button className={"color-" + index} onClick={this.updateCategory}key={index}>{category}</button>
          })
        }
      </div>
    )
  }
}

export default Welcome;