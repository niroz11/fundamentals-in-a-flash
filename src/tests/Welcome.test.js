import React from 'react';
import Welcome from '../Welcome';
import { shallow } from 'enzyme';

const updateCategoryMock = jest.fn();
const categoriesMock = ["Prototype Methods", "OOP", "Scope"];

describe('Welcome', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Welcome
        updateCategory={updateCategoryMock}
        categories={categoriesMock}
      />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update category when a category is clicked', () => {
    wrapper.find('.color-0').simulate('click', { preventDefault: () => { }, target: categoriesMock[0]});
    expect(updateCategoryMock).toBeCalled();
  });

});