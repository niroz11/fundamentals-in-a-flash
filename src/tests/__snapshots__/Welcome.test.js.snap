// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should match the snapshot with all data passed in 1`] = `
<div
  className="welcome"
>
  <p>
    Take these short quizzes to solidify your JS Fundamentals knowledge
  </p>
  <p>
    Select a category below to get started
  </p>
  <button
    className="category-btn-prototype-methods"
    key="0"
    onClick={[Function]}
  >
    Prototype Methods
  </button>
  <button
    className="category-btn-oop"
    key="1"
    onClick={[Function]}
  >
    OOP
  </button>
  <button
    className="category-btn-scope"
    key="2"
    onClick={[Function]}
  >
    Scope
  </button>
</div>
`;
